submissions/Roophee/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+class Inhabitant  {
+    constructor(name, gender, saying, legs=4, tail= 1 ){
+        this.name = name;
+        this.gender = gender;
+        this.species = this.constructor.name.toLowerCase();
+        this.saying = saying;
+        this.legs = legs;
+        this.tail = tail;
+        this.friends = [];
+    }
+
+    addFriends(...arrayOfFriends) {
+        arrayOfFriends.forEach(({name}) => this.friends.push(name));
+    }
+
+    description(){
+        let descriptionArray = ['name: '+ this.name,
+            'gender: '+ this.gender,
+            'species: '+ this.species,
+            'saying: '+ this.saying,
+            'legs: '+ this.legs,
+            'tail: '+ this.tail,
+            'friends: '+ (this.friends.length === 0? 'none' : this.friends),
+        ];
```

Comment:
Example:
```
['propA', 'propB'].map(propName => `${propName} = ${obj[propName}`)
```

You may still need to add handling of array props in the map callback. Trust you can. Hint: a ternary in place of `` ${obj[propName} `` with a condition `Array.isArray(x) && x.length === yyy`

Reply:
Don't understand what for it needs?
You mean I should rewrite method that create string of properties and it had to create string not in way ofnamed enumeration of properties, but in way of automatic enumeration of properties by Array methods?

Right?

Reply:
Right. Imagine you need to add dozens of props in the code base created by someone else (other you)

Reply:
Done. Rewrite description method and this way more universal. That's why there is  no need to override the description method in inheriting objects.

Reply:
Universal methods are often error prone and do not add to readability. Explcitness should win above implicitness in the most of cases. Take care of peer devs that may have to extend or amend functionality of your code.

Reply:
Can I use:
 for ( let property in obj) {...} ?